serde_derive/src/de.rs

```diff
@@ -268,7 +268,11 @@ fn deserialize_in_place_body(cont: &Container, params: &Parameters) -> Option<St
 
     let code = match cont.data {
         Data::Struct(Style::Struct, ref fields) => {
-            deserialize_struct_in_place(None, params, fields, &cont.attrs, None)
+            if let Some(code) = deserialize_struct_in_place(None, params, fields, &cont.attrs, None) {
```

Comment:
Yep, but as far as I can tell serde does not use `?` in the codebase.

Reply:
Yeah it showed up noticeably in benchmarks last time I tried it -- https://github.com/rust-lang/rust/issues/37939. It would be fine to use `?` in serde_derive but not in serde or the generated code, but it has just been simpler to use `try!` everywhere until the performance issue is figured out. That way we never have to argue about whether a particular use of `?` ever matters for performance.